### PR TITLE
Fix placeholders for ignore

### PR DESCRIPTION
**Pull request by frombetelgeuse**
What is done:
- Slightly simplified interface of FileHelper (It is not necessary to solution)
- Added  class to placeholder functionality
- Inserted it where necessary to solve the problem
- Added some tests

### CODE REVIEW

File: src/main/java/com/crowdin/cli/utils/PlaceholderUtil.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    private List<Language> supportedLangs;
+    private List<Language> projectLangs;
+    private String basePath;
+
+    public PlaceholderUtil(List<Language> supportedLangs, List<Language> projectLangs, String basePath) {
+        if (supportedLangs == null || projectLangs == null || basePath == null)
+            throw new NullPointerException("in PlaceholderUtil.contructor");
+        for(Language projectLang : projectLangs) {
+            if (!supportedLangs.parallelStream().anyMatch(lang -> lang.getName().equals(projectLang.getName()))) {
```
Comment (yevheniyJ) [inline src/main/java/com/crowdin/cli/utils/PlaceholderUtil.java]: outer foreach can be replaced into something like this:
`projectLangs.stream()
  .findFirst(projectLang -> !supportedLangs.stream().anyMatch(lang -> lang.getName().equals(projectLang.getName())))
  .ifPresent(projectLang -> throw new RuntimeException("Proj contains langs that Crowdin doesn't support: " + projectLang.getName()))`;

File: src/main/java/com/crowdin/cli/utils/PlaceholderUtil.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                throw new RuntimeException("Proj contains langs that Crowdin doesn't support: " + projectLang.getName());
+            }
+        }
+        this.supportedLangs = supportedLangs;
+        this.projectLangs = projectLangs;
+        this.basePath = basePath;
+    }
+
+    public List<String> format(List<File> sources, List<String> toFormat, boolean onProjectLangs) {
+        if (sources == null || toFormat == null)
```
Comment (yevheniyJ) [inline src/main/java/com/crowdin/cli/utils/PlaceholderUtil.java]: add brackets

File: src/main/java/com/crowdin/cli/utils/PlaceholderUtil.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            return new ArrayList<>();
+        List<String> res = new ArrayList<>();
+        for(String str : toFormat) {
+            res.addAll(format(sources, str, onProjectLangs));
```
Comment (yevheniyJ) [inline src/main/java/com/crowdin/cli/utils/PlaceholderUtil.java]: please use `this` keyword when invoking instance methods

File: src/main/java/com/crowdin/cli/utils/PlaceholderUtil.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        }
+        return res;
+    }
+
+    public List<String> format(List<File> sources, String toFormat, boolean onProjectLangs) {
+        if (sources == null || toFormat == null)
+            return new ArrayList<>();
+        List<String> result = new ArrayList<>();
+        for(Language lang : (onProjectLangs ? projectLangs : supportedLangs)) {
```
Comment (yevheniyJ) [inline src/main/java/com/crowdin/cli/utils/PlaceholderUtil.java]: Use streams instead of loops. `langs.stream().flatMap(...)`

**Review by yevheniyJ — CHANGES_REQUESTED** [review]

File: src/main/java/com/crowdin/cli/properties/helper/FileHelper.java [hunk 5]
```diff
@@ -95,25 +106,21 @@
      * Filters the provided list of source files using the configured filters.
      *
      * @param sources  the source files.
-     * @param fileBean the file bean from the config.
+//     * @param fileBean the file bean from the config.
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/cli/properties/helper/FileHelper.java]: Please update annotations

**Review by yevheniyJ — APPROVED** [review]

**Review by andrii-bodnar — APPROVED** [review]

**Pull request merged**